### PR TITLE
Adding YAML file to run on multiplexed RPF dataset from Gupta et. al. eLife 2018

### PR DESCRIPTION
**Pull request by acope3**
Please let me know if there are any changes that need to be made to the YAML file.

### CODE REVIEW

**Comment by FlicAnderson** [comment]
Hi, might be worth editing out your username etc from the paths so they're something more like `$HOME/example-datasets/`... ?

Do you have developer access to the `riboviz` and `example-datasets` repositories?  If so, is there any way you can make this an issue branch off example-datasets repo (ie just a branch name with `-17` on the end so it matches to the issue ticket) so I can check it more easily?  I've not cloned a branch of a fork before and I'm not sure what the best way to do that is while getting everything into place so I can run it locally.  I'm having a go though :D :nut_and_bolt: :hammer:

**Comment by FlicAnderson** [comment]
Question: @acope3 what do you find the best way of downloading/obtaining the sample sheet is?  Do you do this with any fancy commandline tools? 

I was just thinking that if we're trying to get :sparkles: automagic :sparkles: SRA file downloads eventually, but we're handling multiplex data and requiring sample sheets, do you think this is something we should consider? @ewallace - any thoughts on this?  

Is it something we might include with the example-dataset annotation & contaminants files, since it's likely to be a small textfile?

Thoughts welcome!

**Comment by ewallace** [comment]
Hi - it's great to have this in so fast!

I agree with @FlicAnderson that it would be great to excise the personal username from the yaml as these should be portable. We are trying to figure out how best to do this in issue #13, in progress.

I think having the branch in a fork is fine, as we can edit and pull request in to the branch by following the "view command line instructions" on the pull request page on github.

**Comment by acope3** [comment]
> Question: @acope3 what do you find the best way of downloading/obtaining the sample sheet is? Do you do this with any fancy commandline tools?
>
> I was just thinking that if we're trying to get  automagic  SRA file downloads eventually, but we're handling multiplex data and requiring sample sheets, do you think this is something we should consider? @ewallace - any thoughts on this?

You bring up an excellent point! Unfortunately, no fancy commandline tools here. The sample sheet was available for download at GEO for this dataset. I don't know if this is the case for all multiplexed datasets. 

 
> Is it something we might include with the example-dataset annotation & contaminants files, since it's likely to be a small textfile?
> 
> Thoughts welcome!

I

**Comment by acope3** [comment]
> > Question: @acope3 what do you find the best way of downloading/obtaining the sample sheet is? Do you do this with any fancy commandline tools?
> >I was just thinking that if we're trying to get  automagic  SRA file downloads eventually, but we're handling multiplex data and requiring sample sheets, do you think this is something we should consider? @ewallace - any thoughts on this?
> 
> You bring up an excellent point! Unfortunately, no fancy commandline tools here. The sample sheet was available for download at GEO for this dataset. I don't know if this is the case for all multiplexed datasets.
> 
> > Is it something we might include with the example-dataset annotation & contaminants files, since it's likely to be a small textfile?
> > Thoughts welcome!
> 

@shahpr Perhaps you know if there are any easy ways to obtain the sample sheets with barcodes for multiplexed data. Is it typical/required to include files with barcodes in multiplexed datasets on GEO?

**Comment by acope3** [comment]
@FlicAnderson I've updated this YAML file. Does this look better?

**Comment by kavousan** [comment]
@FlicAnderson to report on a test run and discuss the YAML changes with @acope3; if all good, we merge and close.

**Comment by FlicAnderson** [comment]
Hi @acope3 -  I was wondering, did you try running this dataset on nextflow at all, or just python?  

I've run into an error running this locally via nextflow and wondered if you'd had the same?

**TL;DR:  seems to be issues with parallel processing in the `mclapply()` function in R in `bamToH5` process when I ran this in nextflow, not sure if it happens in python**

```
(riboviz) fanders6@ceres:~/riboviz/riboviz$ nextflow run prep_riboviz.nf -params-file G-Sc_2018/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config.yaml -ansi-log false
N E X T F L O W  ~  version 20.07.1
Launching `prep_riboviz.nf` [stoic_austin] - revision: e273db8045
[45/f18710] Submitted process > buildIndicesORF (yeast_CDS_w_250)
[42/f80a8b] Submitted process > buildIndicesrRNA (yeast_rRNA)
[fd/63d674] Submitted process > cutAdaptersMultiplex (SRR8375039)
[3e/193f69] Submitted process > extractUmisMultiplex (SRR8375039)
[e5/4e78e8] Submitted process > demultiplex (SRR8375039)
Demultiplexed samples: [WT_1, WT_2, WT_3, ncs2_1, ncs2_2, ncs2_3, uba4_1, uba4_2, uba4_3]
Non-demultiplexed samples: []
[de/4ace53] Submitted process > hisat2rRNA (uba4_2)
[e4/cdcc42] Submitted process > hisat2rRNA (ncs2_3)
[11/e1bd51] Submitted process > hisat2rRNA (uba4_1)
[1f/2884fd] Submitted process > hisat2rRNA (WT_1)
[fe/179146] Submitted process > hisat2rRNA (uba4_3)
[eb/fdd273] Submitted process > hisat2rRNA (ncs2_1)
[7b/822e47] Submitted process > hisat2rRNA (ncs2_2)
[48/919812] Submitted process > hisat2rRNA (WT_2)
[2d/f6bf52] Submitted process > hisat2rRNA (WT_3)
[c1/571f9e] Submitted process > hisat2ORF (uba4_3)
[0d/02a5e8] Submitted process > trim5pMismatches (uba4_3)
[e8/65ee23] Submitted process > samViewSort (uba4_3)
[3e/30ea90] Submitted process > hisat2ORF (ncs2_3)
[27/8ecd56] Submitted process > dedupUmis (uba4_3)
[4c/f78ca8] Submitted process > groupUmisPreDedup (uba4_3)
[5e/0cc4ff] Submitted process > trim5pMismatches (ncs2_3)
[f1/d0d06c] Submitted process > samViewSort (ncs2_3)
[4e/2f5fcf] Submitted process > groupUmisPreDedup (ncs2_3)
[c3/c72ad0] Submitted process > dedupUmis (ncs2_3)
[db/078cf2] Submitted process > hisat2ORF (WT_3)
[39/060ee3] Submitted process > groupUmisPostDedup (ncs2_3)
[c0/05f4d1] Submitted process > outputBams (ncs2_3)
[5c/aa1f08] Submitted process > makeBedgraphs (ncs2_3)
[ef/5ee041] Submitted process > bamToH5 (ncs2_3)
[ee/9648f8] Submitted process > trim5pMismatches (WT_3)
[34/bb23b7] Submitted process > samViewSort (WT_3)
[65/5aef29] Submitted process > groupUmisPreDedup (WT_3)
[79/924d62] Submitted process > dedupUmis (WT_3)
[1b/33585c] Submitted process > outputBams (uba4_3)
[a9/d703a6] Submitted process > groupUmisPostDedup (uba4_3)
[13/a9ae0e] Submitted process > makeBedgraphs (uba4_3)
[2c/d79dec] Submitted process > bamToH5 (uba4_3)
[8a/d1429a] Submitted process > hisat2ORF (ncs2_2)
[04/91c63d] Submitted process > groupUmisPostDedup (WT_3)
[6b/fd0d4c] Submitted process > outputBams (WT_3)
[60/d308e2] Submitted process > bamToH5 (WT_3)
[a1/a7fbde] Submitted process > makeBedgraphs (WT_3)
[ef/5ee041] NOTE: Process `bamToH5 (ncs2_3)` terminated with an error exit status (1) -- Error is ignored
[93/67771f] Submitted process > hisat2ORF (ncs2_1)
[90/9efc6a] Submitted process > hisat2ORF (uba4_1)
[d8/466fdf] Submitted process > trim5pMismatches (ncs2_2)
[e3/4a73b5] Submitted process > samViewSort (ncs2_2)
[21/1dd4e2] Submitted process > groupUmisPreDedup (ncs2_2)
[47/dd55c6] Submitted process > dedupUmis (ncs2_2)
[58/11673f] Submitted process > hisat2ORF (WT_1)
[37/325dca] Submitted process > trim5pMismatches (uba4_1)
[c8/cbb916] Submitted process > trim5pMismatches (ncs2_1)
[a3/87005e] Submitted process > samViewSort (uba4_1)
[f3/1fb4fa] Submitted process > samViewSort (ncs2_1)
[62/8a67f3] Submitted process > hisat2ORF (uba4_2)
[9b/6e2419] Submitted process > groupUmisPreDedup (uba4_1)
[df/919878] Submitted process > dedupUmis (uba4_1)
[f7/412184] Submitted process > trim5pMismatches (WT_1)
[59/ac717b] Submitted process > dedupUmis (ncs2_1)
[be/77bb52] Submitted process > groupUmisPreDedup (ncs2_1)
[91/4976d7] Submitted process > samViewSort (WT_1)
[8d/80fc45] Submitted process > groupUmisPostDedup (ncs2_2)
[b7/7fb579] Submitted process > outputBams (ncs2_2)
[ef/6549fc] Submitted process > makeBedgraphs (ncs2_2)
[00/41b760] Submitted process > bamToH5 (ncs2_2)
[bc/af6da5] Submitted process > groupUmisPreDedup (WT_1)
[8a/f7de8d] Submitted process > dedupUmis (WT_1)
[d5/552d07] Submitted process > trim5pMismatches (uba4_2)
[31/686043] Submitted process > samViewSort (uba4_2)
[84/76dfff] Submitted process > groupUmisPreDedup (uba4_2)
[4c/a44c22] Submitted process > dedupUmis (uba4_2)
[fc/56b6e3] Submitted process > hisat2ORF (WT_2)
[18/cd3f9f] Submitted process > trim5pMismatches (WT_2)
[f2/dbb43f] Submitted process > samViewSort (WT_2)
[d2/32bdac] Submitted process > dedupUmis (WT_2)
[ec/57ebfa] Submitted process > groupUmisPreDedup (WT_2)
[5c/29a19d] Submitted process > groupUmisPostDedup (ncs2_1)
[64/284d0a] Submitted process > outputBams (ncs2_1)
[38/62d29d] Submitted process > bamToH5 (ncs2_1)
[70/bbe2dc] Submitted process > makeBedgraphs (ncs2_1)
[89/3d9433] Submitted process > groupUmisPostDedup (WT_1)
[7b/9108f0] Submitted process > outputBams (WT_1)
[aa/a31f84] Submitted process > makeBedgraphs (WT_1)
[16/d21996] Submitted process > bamToH5 (WT_1)
[d4/dc0597] Submitted process > groupUmisPostDedup (uba4_2)
[a0/e7e7bb] Submitted process > outputBams (uba4_2)
[99/b4c5e8] Submitted process > bamToH5 (uba4_2)
[f5/c30468] Submitted process > makeBedgraphs (uba4_2)
[2c/d79dec] NOTE: Process `bamToH5 (uba4_3)` terminated with an error exit status (137) -- Error is ignored
[60/d308e2] NOTE: Process `bamToH5 (WT_3)` terminated with an error exit status (137) -- Error is ignored
[00/41b760] NOTE: Process `bamToH5 (ncs2_2)` terminated with an error exit status (137) -- Error is ignored
[23/65eff1] Submitted process > groupUmisPostDedup (uba4_1)
[74/0b87a5] Submitted process > outputBams (uba4_1)
[a9/143e2c] Submitted process > bamToH5 (uba4_1)
[8c/1a661c] Submitted process > makeBedgraphs (uba4_1)
[38/62d29d] NOTE: Process `bamToH5 (ncs2_1)` terminated with an error exit status (1) -- Error is ignored
[16/d21996] NOTE: Process `bamToH5 (WT_1)` terminated with an error exit status (1) -- Error is ignored
[99/b4c5e8] NOTE: Process `bamToH5 (uba4_2)` terminated with an error exit status (1) -- Error is ignored
[91/603394] Submitted process > groupUmisPostDedup (WT_2)
[fd/bc1254] Submitted process > outputBams (WT_2)
[0b/c3b045] Submitted process > makeBedgraphs (WT_2)
[6e/1d427e] Submitted process > bamToH5 (WT_2)
[a9/143e2c] NOTE: Process `bamToH5 (uba4_1)` terminated with an error exit status (1) -- Error is ignored
[02/ebb431] Submitted process > generateStatsFigs (WT_2)
Finished processing sample: WT_2
[3b/03cd13] Submitted process > renameTpms (WT_2)
[c1/50afc4] Submitted process > collateTpms (WT_2)
[e4/4eacf7] Submitted process > countReads
Workflow finished! (OK)
```
So I seem to have had some successes (ie `generateStatsFigs` process has completed and made plot outputs for sample `WT_2`, but none of the others I think?), but quite a few issues with the `bamToH5` process.  If the `bamToH5` process fails, it won't be creating an `[sample].H5` file required by `generateStatsFigs`, so this makes sense. 

The errors on the `bamToH5` processes are interesting... I checked the `.command.err` file for one of them:
```
(riboviz) fanders6@ceres:~/riboviz/riboviz$ cat work/a9/143e2cd7e2f0533637863a5e815f83/.command.err
The following object is masked _by_ .GlobalEnv:
    secondary_id

 *** caught segfault ***
address 0x111000005, cause 'memory not mapped'

Traceback:
 1: factor(read_loc_len_i, levels = nucleotide_pos)
 2: eval(quote(list(...)), env)
 3: eval(quote(list(...)), env)
 4: eval(quote(list(...)), env)
 5: standardGeneric("table")
 6: table(factor(read_loc_len_i, levels = nucleotide_pos))
 7: reads_to_list(gene_location = gene_location, bam_file = bam_file,     read_range = read_range, left_flank = Buffer_l, right_flank = Buffer_r,     mult_exon = TRUE)
 8: FUN(X[[i]], ...)
 9: lapply(X = S, FUN = FUN, ...)
10: doTryCatch(return(expr), name, parentenv, handler)
11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
12: tryCatchList(expr, classes, parentenv, handlers)
13: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        sm <- strsplit(conditionMessage(e), "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && isTRUE(getOption("show.error.messages"))) {        cat(msg, file = outFile)        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
14: try(lapply(X = S, FUN = FUN, ...), silent = TRUE)
15: sendMaster(try(lapply(X = S, FUN = FUN, ...), silent = TRUE))
16: FUN(X[[i]], ...)
17: lapply(seq_len(cores), inner.do)
18: mclapply(genes, function(gene) {    gene_location = gff[gff_pid == gene]    geneseqname <- as.character(seqnames(gene_location)[1])    seqlevels(gene_location) <- geneseqname    if (is_riboviz_gff) {        Buffer_l <- width(gene_location[gene_location$type ==             "UTR5"])        Buffer_r <- width(gene_location[gene_location$type ==             "UTR3"])        gene_location <- gene_location[gene_location$type ==             "CDS"]    }    else {        Buffer_l <- buffer        Buffer_r <- buffer    }    reads_to_list(gene_location = gene_location, bam_file = bam_file,         read_range = read_range, left_flank = Buffer_l, right_flank = Buffer_r,         mult_exon = TRUE)}, mc.cores = num_processes)
An irrecoverable exception occurred. R is aborting now ...
Warning message:
In mclapply(genes, function(gene) { :
  scheduled core 5 did not deliver a result, all values of the job will be affected
Error in apply(output, 1, sum) : dim(X) must have a positive length
Calls: h5writeAttribute.integer -> h5writeAttribute.array -> apply
Execution halted
```
..."Segfault" sounds fairly ominous? Likewise: "An irrecoverable exception occurred. R is aborting now ..." & "In mclapply(genes, function(gene){ scheduled core 5 did not deliver a result, all values of the job will be affected"

For another of the `bamToH5` processes I get similar issues. Here's my `.command.log` and `.command.err`: 
```
(riboviz) fanders6@ceres:~/riboviz/riboviz$ cat work/16/d21996def6a99220e4a9e5a9520d28/.command.log
[1] "Created by: RiboViz"
[1] "Date: 2020-10-10 16:55:24"
[1] "File: /home/fanders6/riboviz/riboviz/rscripts/bam_to_h5.R"
[1] "Version: commit e176f0b13b69a2bd584b4c54bd900c6f2cee7008 date 2020-10-05 17:11:10 GMT"
The following object is masked _by_ .GlobalEnv:
    secondary_id
[1] "bam_to_h5.R running with parameters:"
$num_processes
[1] 16
$min_read_length
[1] 10
$max_read_length
[1] 50
$buffer
[1] 250
$primary_id
[1] "Name"
$secondary_id
[1] "NULL"
$bam_file
[1] "WT_1.bam"
$hd_file
[1] "WT_1.h5"
$orf_gff_file
[1] "Saccharomyces_cerevisiae_yeast_CDS_w_250utrs.gff3"
$dataset
[1] "G-Sc_2018"
$stop_in_cds
[1] FALSE
$is_riboviz_gff
[1] TRUE
$is_test_run
[1] FALSE
$help
[1] FALSE
[1] "Mapping reads to CDS"
Warning message:
In mclapply(genes, function(gene) { :
  scheduled cores 1, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16 did not deliver results, all values of the jobs will be affected
[1] "Saving mapped reads in a H5 file"
[1] TRUE
Error in apply(output, 1, sum) : dim(X) must have a positive length
Calls: h5writeAttribute.integer -> h5writeAttribute.array -> apply
Execution halted

# .command.err
(riboviz) fanders6@ceres:~/riboviz/riboviz$ cat work/16/d21996def6a99220e4a9e5a9520d28/.command.err
The following object is masked _by_ .GlobalEnv:
    secondary_id
Warning message:
In mclapply(genes, function(gene) { :
  scheduled cores 1, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16 did not deliver results, all values of the jobs will be affected
Error in apply(output, 1, sum) : dim(X) must have a positive length
Calls: h5writeAttribute.integer -> h5writeAttribute.array -> apply
Execution halted
```
I haven't run into `"scheduled cores did not deliver results"` errors before, this is new for me. I'm curious if this is something where my laptop was just having a bad day, or something you can recreate?

Stack Overflow has [a bit about this error](https://stackoverflow.com/questions/62811745/scheduled-cores-did-not-deliver-results-all-values-of-the-jobs-will-be-affe) and this suggests that the `mclapply()` maybe doesn't play well with multithreading, and then I fell down a rabbit hole of [what that is](https://stackoverflow.com/a/14190845) and whether my Intel i9 processor does it, and if this is an issue.  

I'm keen to re-try this dataset again a) with the python call, b) on Eddie, which apparently doesn't do 'hyperthreading' but I don't know if it does 'multithreading', honestly this parallel stuff is a little beyond me currently. I still need to check if I can now get Eddie working after [riboviz/#220](https://github.com/riboviz/riboviz/issues/220). 

@mikej888 have you come across anything like this before?

**Comment by acope3** [comment]
@FlicAnderson No, I forgot to do this on my end and was very appreciative of you taking a shot at it. I ran it with Nextflow on my end and I'm having the same problem. Weird. I have no idea what could be causing this.

**Comment by acope3** [comment]
@FlicAnderson I hadn't run it on nextflow because I forgot to, so I was very appreciative when you took the initiative to do it last week. I ran it yesterday and it looks like `generateStatsFigs.R` completed for all of the samples in multiplexed data. Have you pulled the change to `sample_sheets.py`? I don't know why that would be the cause of the issue, but that's potentially one difference between my local riboviz and your version.

**Comment by mikej888** [comment]
@FlicAnderson I haven't seen that error before. A segmentation fault often can arise when a program tries to access a part of memory it isn't allowed to or shouldn't (very common when writing C). Multithreading is typically when a program runs tasks concurrently - this may not necessarily be in parallel as even a single processor can execute a multithreaded program.

I Googled "mclapply address cause memory not mapped" and got another Stackoverflow, 
[caught segfault - 'memory not mapped' error in R](https://stackoverflow.com/questions/49190251/caught-segfault-memory-not-mapped-error-in-r) which has various possibilities including "a mismatch between currently installed shared libraries and the libraries that were built to install R or packages".

**Comment by ewallace** [comment]
We discussed this at 12 Oct 2020 dev meeting. Still waiting for @FlicAnderson or someone to try again and see if we can reproduce the bug, or if it runs. @acope3 has added the barcodes/sample sheet to the yaml file, and has run this again on nextflow (on the shah lab cluster) and it works there for him.

**Comment by FlicAnderson** [comment]
I still need to re-run this to check if I can reproduce the bug I got earlier.  I'll make sure I pull changes first as @acope3 suggests!  Also will try running this on Eddie to see if that works now that's up again (after [riboviz/#220](https://github.com/riboviz/riboviz/issues/220) ).

Aiming to try and do this before Tues 3rd Nov Dev meeting, but I'm not sure I'll have time to get around to it. Sorry for the delay @acope3 and other folks!

**Comment by FlicAnderson** [comment]
Hi @acope3 I've submitted the job on Eddie cluster now, so hopefully (unless I've done anything daft in the setup) that should run. I'm not sure it'll run before our meeting though, sorry :S  I haven't had time to queue it up until now. 

One thing that might be worth editing before accepting the PR: 
 - in the yaml, remove the `>` from the 'notes' section of the provenance info
 - edit `riboviz-version` to 2.0 (not beta) and update the commit if it's not updated?

Re: the `>` in 'notes: it doesn't add anything, and I think that for some reason including that seems to crash runs on Eddie, so I need to manually edit it out.  I need to make sure I've changed this in any of our existing example-datasets and wherever we put the details on what provenance info to include. Give me a shout if you see it lurking anywhere else!

**Comment by FlicAnderson** [comment]
I've just resubmitted this to Eddie for what feels like the umpteenth time - sorry @acope3, I've had a few daft path errors and then a couple of resource-related job aborts. 

I had run it with these grid engine options: 
```
#$ -l h_rt=8:00:00 
#$ -l h_vmem=4G 
#$ -pe mpi 16
#$ -R y
```
But sadly it exceeded 64GB during one of the processes & got killed. Resubmitted using `-pe mpi 32` instead, but that means another long wait until it runs. I was definitely hoping I could say "yep, all fine!" and you could close and merge this today...

**Comment by FlicAnderson** [comment]
... I used `-resume` function by updating the nextflow command on the job submission script while it was still in the queue, and it's now running again and has resumed from where it left off the last time. Currently at the `extractUmisMultiplex (SRR8375039)` process! :fast_forward:  :crossed_fingers:

**Comment by FlicAnderson** [comment]
Success! Finally got your PR run on Eddie via Nextflow @acope3!

For future reference, the successful combo of grid engine resources that let the job actually complete without getting a memory-based kill was this:
```
#$ -l h_rt=8:00:00 
#$ -pe mpi 32
#$ -R y
#$ -l h_vmem=8G
``` 
I had to boost the h_vmem to 8Gb instead of 4Gb. But that's it fine now. 

All of the 9 samples produced plot outputs, and they looked pretty good. Here's the `head` and `tail` of  `TPMs_collated.tsv` and the file sizes of the contents of the `output` directory, in case you've still got your outputs for comparison?

```
[fanders6@login01(eddie) output]$ head TPMs_collated.tsv 
# Created by: RiboViz
# Date: 2020-11-19 16:57:07
# File: /exports/eddie3_homes_local/fanders6/riboviz/riboviz/rscripts/collate_tpms.R
# Version: commit 35ee466e45a415f5bee23e90b2d864f972fe7285 date 2020-11-03 15:14:12 GMT
ORF	ncs2_3	WT_3	uba4_3	ncs2_2	uba4_2	WT_1	ncs2_1	WT_2	uba4_1
Q0045	2	5.8	2.3	1.6	1.9	1.8	1	0.9	1.6
Q0050	6.5	1.9	0.5	2.6	2.1	1.9	1.2	2.3	0.3
Q0055	6.3	0.9	3	1.5	0.4	1.5	1.5	2	0.2
Q0060	2.6	0	0	1	0.8	0.7	0	1.1	1
Q0065	0	0	0.7	0.8	0	0.6	0.9	0	0.4

[fanders6@login01(eddie) output]$ tail TPMs_collated.tsv 
YPR193C	0	9.2	9.9	7.7	8.1	13.1	9.2	9.9	11.3
YPR194C	7.4	8.9	5.8	4.5	4.7	10.1	8.6	8.2	5.3
YPR196W	9.1	6.6	7.9	9.1	11.5	4.7	13	8.1	12
YPR198W	13.8	17.1	14.5	13.5	15.6	17.3	20.3	12.7	18.6
YPR199C	0	5.1	9.7	14.3	10.1	3.1	8.5	7.1	6.3
YPR200C	0	0	0	0	0	0	0	0	0
YPR201W	0	1.9	0	0	0.8	0	0.6	0	0.5
YPR202W	0	3.1	1.7	3.5	2.7	5.1	3.1	1.9	0.8
YPR203W	0	6.7	0	0	2.9	0	2.2	4.1	1.8
YPR204W	1.1	0	1.6	0.4	0.3	0.3	0.3	0.2	1.4

[fanders6@login01(eddie) G-Sc_2018]$ du -ch output/
104M	output/ncs2_1
114M	output/uba4_1
71M	output/WT_3
86M	output/ncs2_2
93M	output/uba4_2
95M	output/WT_1
105M	output/WT_2
67M	output/ncs2_3
86M	output/uba4_3
816M	output/
816M	total
```
Downloaded the outputs to my comp and the plots do look fine, with pretty good periodicity, read lengths, and pretty high probability of frame0. 

Nice one.  If you're happy that this sounds like what you've got, then I'd say go for the merge!
